cmd/tailscale/cli/admin.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Admin commands.
+
+package cli
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"tailscale.com/client/tailscale"
+)
+
+var adminCmd = &ffcli.Command{
+	Name:     "admin",
+	Exec:     runAdmin,
+	LongHelp: `"tailscale admin" contains admin commands to manage a Tailscale network.`,
+	FlagSet: (func() *flag.FlagSet {
+		fs := newFlagSet("admin")
+		fs.StringVar(&adminArgs.apiBase, "api-server", "https://api.tailscale.com", "which Tailscale server instance to use. Ignored when --token-file is empty.")
+		fs.StringVar(&adminArgs.tokenFile, "token-file", "", "if non-empty, filename containing API token to use. If empty, authentication is done via the active Tailscale control plane connection.")
+		fs.StringVar(&adminArgs.tailnet, "tailnet", "", "Tailnet to query or edit. Required if token-file is used. Must be blank if token-file is blank, in which case the tailnet used is the same as the active tailnet.")
+		return fs
+	})(),
+	Subcommands: []*ffcli.Command{
+		{
+			Name:      "list-devices",
+			Exec:      runAdminListDevices,
+			ShortHelp: "list devices",
+		},
+	},
+}
+
+var adminArgs struct {
+	tokenFile string
+	tailnet   string
+	apiBase   string
+}
+
```

Comment:
let's make this an actual client library.

Reply:
For sure but not yet.

Reply:
when it's a real client library please make it replace the partially-done one we have in corp

Reply:
The problem is nobody wants a client library in oss with our massive go.mod and bleeding edge Go dependency.

So a client library really needs to live elsewhere and work on older Gos, so I don't intend to do it in this commit (as it's the wrong repo)

The separate repo would probably then need to be like https://github.com/tailscale/tscert which we made for Caddy, which includes a thinned down version of the client/tailscale package for speaking to the LocalAPI.  (as we'll want the real control API to use Local API for the Noise proxying)

Reply:
Shouldn't [lazy module loading](https://go.dev/ref/mod#lazy-loading) make our go.mod size irrelevant? it wouldn't fix the older Go version problem though.